tests/ui/GroupChatNameTests.tsx

```diff
@@ -0,0 +1,420 @@
+ * The reason we need this is because we need to trigger the transitionEnd event in our tests to simulate
+ * the transitionEnd event that is triggered when the screen transition animation is completed.
+ *
+ * P.S: This can't be moved to a utils file because Jest wants any external function to stay in the scope.
+ *
+ * @returns An object with two functions: triggerTransitionEnd and addListener
+ */
+const createAddListenerMock = (): ListenerMock => {
+    const transitionEndListeners: Array<() => void> = [];
+    const triggerTransitionEnd = () => {
+        transitionEndListeners.forEach((transitionEndListener) => transitionEndListener());
+    };
+
+    const addListener: jest.Mock = jest.fn().mockImplementation((listener, callback) => {
+        if (listener === 'transitionEnd') {
+            transitionEndListeners.push(callback);
+        }
+        return () => {
+            // eslint-disable-next-line rulesdir/prefer-underscore-method
+            transitionEndListeners.filter((cb) => cb !== callback);
+        };
+    });
+
+    return {triggerTransitionEnd, addListener};
+};
+
+jest.mock('@react-navigation/native', () => {
+    const actualNav = jest.requireActual('@react-navigation/native');
+    const {triggerTransitionEnd, addListener} = createAddListenerMock();
+    transitionEndCB = triggerTransitionEnd;
+
+    const useNavigation = () =>
+        ({
+            navigate: jest.fn(),
+            ...actualNav.useNavigation,
+            getState: () => ({
+                routes: [],
+            }),
+            addListener,
+        } as typeof NativeNavigation.useNavigation);
+
+    return {
+        ...actualNav,
+        useNavigation,
+        getState: () => ({
+            routes: [],
+        }),
+    } as typeof NativeNavigation;
+});
+
+beforeAll(() => {
+    // In this test, we are generically mocking the responses of all API requests by mocking fetch() and having it
+    // return 200. In other tests, we might mock HttpUtils.xhr() with a more specific mock data response (which means
+    // fetch() never gets called so it does not need mocking) or we might have fetch throw an error to test error handling
+    // behavior. But here we just want to treat all API requests as a generic "success" and in the cases where we need to
+    // simulate data arriving we will just set it into Onyx directly with Onyx.merge() or Onyx.set() etc.
+    // @ts-expect-error -- TODO: Remove this once TestHelper (https://github.com/Expensify/App/issues/25318) is migrated
+    global.fetch = TestHelper.getGlobalFetchMock();
+
+    Linking.setInitialURL('https://new.expensify.com/');
+    appSetup();
+});
+
+async function navigateToSidebarOption(index: number): Promise<void> {
+    const hintText = Localize.translateLocal('accessibilityHints.navigatesToChat');
+    const optionRows = screen.queryAllByAccessibilityHint(hintText);
+    fireEvent(optionRows[index], 'press');
+    await waitForBatchedUpdatesWithAct();
+}
```

Comment:
Feels like we copied this from `UnreadIndicatorsTest`. Can we create some utility method instead of copying and pasting?